Clean up LBFGS

vladimir-ch opened the pull request:
@btracey PTAL

btracey commented on lbfgs.go:
```diff
@@ -1,52 +1,61 @@
-// Since BFGS has a cost of O(inputDimension^2), LBFGS is more appropriate
-// for very large problems. This "forgetful" nature of LBFGS may also make it perform
-// better than BFGS for functions with Hessians that vary rapidly spatially.
+// LBFGS implements the limited-memory BFGS optimization method.
```
why method instead of algorithm?

vladimir-ch replied:
Just for consistency with the other optimizers which all use the word "method".

btracey commented on lbfgs.go:
```diff
@@ -1,52 +1,61 @@
+//
+// While the normal BFGS algorithm makes a full approximation to the inverse
+// Hessian, LBFGS instead approximates the Hessian from the last Store
+// iterations. The Store parameter is a tradeoff between cost of the method and
```
between storage cost. Strictly it's both, but storage is the important dimension. We can leave the last line to highlight it's both.

btracey replied:
I guess that's not true. I would delete this last sentence here and move the whole discussion into the next paragraph. This section says what it does and the next one says why

btracey commented on lbfgs.go:
```diff
@@ -1,52 +1,61 @@
-// If Store is 0, Store is defaulted to 15.
-// A Linesearcher for LBFGS must satisfy the strong Wolfe conditions at every
-// iteration. If Linesearcher == nil, an appropriate default is chosen.
+// Since BFGS has a cost of O(dimension^2), LBFGS is more appropriate for very
```
I would flip the sentence

LBFGS is more appropriate than BFGS for large dimensional problems as the cost of LBFGS scales as O(Store * dim) while BFGS scales with O(dim^2).

btracey commented on lbfgs.go:
```diff
@@ -1,52 +1,61 @@
 	Linesearcher Linesearcher
-	Store        int // how many past iterations to store
+	// Store is the size of the limited-memory storage.
+	// If Store is 0, it will be defaulted to 15.
```
We should choose one of is and ==. @kortschak  the preferred is english in comments, right?

kortschak replied:
English over symbols if possible.

btracey commented:
LGTM

vladimir-ch commented:
@btracey Could you please take one last look? I changed the LBFGS comment and I deleted the tradeoff comment. I considered deleting the word "spatially" because from the point of view of LBFGS there is no other than spatial variation but in the end I left it there.

btracey commented:
LGTM
